common-app-lib/src/cmr/common_app/api/log.clj

```diff
@@ -0,0 +1,133 @@
+    namespace to debug and all other names spaces are set to warn.
+    ******* Important! - The main logging level :level - must be set at the lowest setting that you
+    want logged at the namespace level. Otherwise Timbre will filter these logs out before the logs
+    make it to the middleware.  The middleware :all will set the logging level for all namespaces
+    not listed - set this to what :level used to be. *****
+    The change command will return the new logging configuration."
+  (:require
+   [cmr.acl.core :as acl]
+   [cheshire.core :as json]
+   [clojure.edn :as edn]
+   [clojure.pprint :as pp]
+   [clojure.string :as string]
+   [cmr.common.log :as log]
+   [cmr.common-app.api.routes :as cr]
+   [cmr.common.mime-types :as mt]
+   [compojure.core :refer :all]))
+
+(defn get-log-configuration-response
+  "Creates a successful log configuration response which is an output of the logging configuration map.
+   This function takes in three parameters, the first is a boolean - if true the output is in html format
+   otherwise the output will be a string representation of the configuration map. The second parameter is
+   the status code and the third is the actual configuration map."
+  [html? status-code data]
+  (let [body-string (if html?
+                      (str "<html><pre><code>" (with-out-str (pp/pprint data)) "</code></pre></html>")
+                      (with-out-str (pp/pprint data)))
+        content (if html?
+                  {cr/CONTENT_TYPE_HEADER mt/html}
+                  {cr/CONTENT_TYPE_HEADER mt/edn})]
+    {:status status-code
+     :body body-string
+     :headers content}))
+
+(defn get-log-error-response
+  "Creates an error response if the content type is not correctly set. If it is not correctly set
+   or not set at all, then the compojure wrap-params will consume the body and it will be blank.
+   Therefore I will send back an error to correctly set the content-type. Otherwise I will have to
+   save the body in another key (like search is doing) in all services and use that key instead."
+  [status-code content-type]
+  {:status status-code
+   :body (str "The content type of "
+              content-type
+              " in the request is either not provided or is not of the correct type. Please set the request Content-Type to application/edn.")
+   :headers {cr/CONTENT_TYPE_HEADER "text/plain"}})
+
+(defn get-logging-configuration
+  "This function gets the logging configuration map to the caller. If a token is not passed in
+   with the request or the user doesn't have permission, the acl will through an exception. The
+   exception is caught in base code and an appropriate error message is passed to the end user."
+  [request-context headers]
+  (acl/verify-ingest-management-permission request-context :update)
+  (get-log-configuration-response
+    (.contains (mt/get-header headers "accept") "text/html")
+    200
+    (log/get-logging-configuration)))
+
+(defn merge-log-configuration
```

Comment:
This is a side-effect-y operation ... it's setting data and not returning anything, so you should change the name with an appended `!`: `merge-logging-configuration!` . (I think `logging` is a better name here, and it matches your function above, `get-logging-configuration`.)

Reply:
In this function I am setting the configuration and then I am returning the new configuration.  Since I am doing that should I still append the function with !?

Reply:
From a reply in chat, you were OK not adding ! since the new configuration is returned.  I have also documented the fact that the function also returns the new configuration.